### DISCUSSION TITLE
TLS support for local server

### DISCUSSION THREAD

**Discussion by danielperna84** [SukramJ/aiohomematic#860] [Ideas]
The local server should allow to enable encryption using certificates. It remains to be tested how the CCU handles this if a self-signed (or otherwise invalid) certificate is being used.

**Comment by SukramJ** [comment]
SimpleXMLRPCServer does not currently support TLS. There are examples from 2012/2013 in which people try to recreate something like this, but there is no current standard implementation (at least I didn't find one).
Even if there was an implementation, certificates would have to be included. This may require instructions on how to use Letsencrypt or how to create and integrate certificates.
Furthermore, the configuration would have to be expanded, and I can well imagine that this would also increase the support effort.
In short: I think the costs are much higher than the benefits.
I would close the ticket.

**Comment by danielperna84** [comment]
I agree. I would leave the issue open though, because sooner or later TLS nerds will show up and ask for this. This issue could then serve as a response to their request. As an alternative we could have a list of features that won't be integrated in the Wiki. But I assume only few would ever look at that before opening an issue. For now I have added the _wontfix_ label.
